src/App.js

```diff
@@ -1,13 +1,28 @@
-
 import './App.css';
 import EmailCardList from "./EmailCardList/EmailCardList.js";
 import Header from './Header/Header.js'
 import SideBar from './SideBar/SideBar.js'
-import {useEffect, useState} from "react"
+import ReadingPanel from "./ReadingPanel/ReadingPanel";
+import {useState, useEffect} from "react";
 
 const App = () => {
     const [allEmailSnippets, setAllEmailSnippets] = useState([])
+    const [emailSearchId, setEmailSearchId] = useState([])
+    const [readingPanelEmailData, setReadingPanelEmailData] = useState([])
     const [sidebarIsHidden, setSidebarIsHidden] = useState(true)
+    const [isReadingPanelOpen, setIsReadingPanelOpen] = useState(false)
+
+    const fetchEmailById = async () => {
+        let fetchUrl = 'http://localhost:8080/emails/' + emailSearchId
+        const emailData = await fetch(fetchUrl)
+        const jsonEmailData = await emailData.json()
+        setReadingPanelEmailData(jsonEmailData.data.email)
+    }
+
+    useEffect(() => {
+        fetchEmailById()
+    }, [emailSearchId])
+
```

Comment:
still need to put both function definitions together and both `useEffect`s together to make the file tidier